### PR TITLE
ACL-8 Form Experimentation

### PR DESCRIPTION
**Pull request by ElGrecode**
### Links
https://autoclub.atlassian.net/browse/ACL-8

### Description
This PR attempts to do 3 things:
1. Adds a high level Form wrapper (`react-final-form`)
2. Provides a library for validation (`utilities/validate.js`)
3. Provides a pathway to "wrap" our base Form components. This wrapper provides the logic for switching error values on/off.
- Each base Form building block will map to a Form wrapper of that. (i.e. `Input.js -> FormInput.js`, `NumericalStepper.js -> FormNumericalStepper.js`)
- This wrapper will provide the logic of when to show errors and passing the value of that form object to the component itself

### How to test
Check out Form Story to see a working example

### Additional Notes
This PR is a discussion on how Forms would potentially function. If we agree that this is the way forward, we can start to create tickets for wrapper components around our form base components.
Check out https://github.com/final-form/react-final-form form more information

### Explanation of Jest/Unit Testing
None yet, this is an RFC

### CODE REVIEW

**Comment by ElGrecode** [comment]
Screen Shot 2019-05-14 at 7 52 33 PM

**Comment by ElGrecode** [comment]
Screen Shot 2019-06-03 at 11 06 54 AM

File: src/lib/utilities/validate.js [hunk 1]
```diff
@@ -0,0 +1,311 @@
...
+
+    return (value ? (`${value}`) : '').length <= parseInt(length, 10);
+  },
+
+  exact_length(value, length) {
+    if (!Validate.regex.integerRegex.test(length)) {
+      return false;
+    }
+
+    return ((value || '').length === parseInt(length, 10));
```
Comment (aaa-magenic-tedb) [inline src/lib/utilities/validate.js]: Looks clean although I wonder if you would consider using logical && operator so there is just one return.  Same comment for the rest that has the same code structure.

**Comment by ElGrecode** [comment]
Closing. I will reopen when there are unit tests and some polish to the overall feel.

**Pull request closed without merging**